### PR TITLE
Fix #109 bug transaction concurrency.

### LINKED ISSUE #109: TypeError: Object [object Object] has no method 'release'
I'm having this error when issuing a `SELECT` and then doing a `#rollback` to close the transaction without actually doing other operation. 

```
2015-09-24T18:31:11.750Z pid:2371 worker:36 Thu, 24 Sep 2015 18:31:11 GMT loopback:connector:postgresql SQL: SELECT "field","id" FROM "schema"."table" WHERE "field"=$1 ORDER BY "id"
2015-09-24T18:31:11.750Z pid:2371 worker:36 Parameters: ["foo"]
2015-09-24T18:31:11.750Z pid:2371 worker:36 Thu, 24 Sep 2015 18:31:11 GMT loopback:connector:postgresql Execute SQL within a transaction
2015-09-24T18:31:11.752Z pid:2371 worker:36 events.js:72
2015-09-24T18:31:11.752Z pid:2371 worker:36         throw er; // Unhandled 'error' event
2015-09-24T18:31:11.752Z pid:2371 worker:36               ^
2015-09-24T18:31:11.754Z pid:2371 worker:36 TypeError: Object [object Object] has no method 'release'
2015-09-24T18:31:11.754Z pid:2371 worker:36     at PostgreSQL.releaseConnection (/home/myapp/node_modules/loopback-connector-postgresql/lib/transaction.js:70:14)
2015-09-24T18:31:11.754Z pid:2371 worker:36     at null.<anonymous> (/home/myapp/node_modules/loopback-connector-postgresql/lib/transaction.js:56:12)
2015-09-24T18:31:11.755Z pid:2371 worker:36     at Object.forward (/home/ubuntu/.nvm/v0.10.38/lib/node_modules/strongloop/node_modules/strong-agent/lib/proxy.js:79:23)
2015-09-24T18:31:11.755Z pid:2371 worker:36     at eval (eval at wrap (/home/ubuntu/.nvm/v0.10.38/lib/node_modules/strongloop/node_modules/strong-agent/lib/proxy.js:193:20), <anonymous>:3:21)
2015-09-24T18:31:11.755Z pid:2371 worker:36     at null.<anonymous> (/home/myapp/node_modules/loopback/node_modules/continuation-local-storage/context.js:74:17)
2015-09-24T18:31:11.755Z pid:2371 worker:36     at b [as callback] (domain.js:183:18)
2015-09-24T18:31:11.755Z pid:2371 worker:36     at Query.handleReadyForQuery (/home/myapp/node_modules/loopback-connector-postgresql/node_modules/pg/lib/query.js:89:10)
2015-09-24T18:31:11.755Z pid:2371 worker:36     at null.<anonymous> (/home/myapp/node_modules/loopback-connector-postgresql/node_modules/pg/lib/client.js:163:19)
2015-09-24T18:31:11.755Z pid:2371 worker:36     at emit (events.js:117:20)
2015-09-24T18:31:11.755Z pid:2371 worker:36     at Socket.<anonymous> (/home/myapp/node_modules/loopback-connector-postgresql/node_modules/pg/lib/connection.js:109:12)
```

Please notice this is only happening in production having the same modules version as in development:

```
npm list loopback loopback-datasource-juggler loopback-connector-postgresql
├── loopback@2.22.2
├── loopback-connector-postgresql@2.3.0
└── loopback-datasource-juggler@2.40.1
```

### PR DESCRIPTION
**Pull request by robinbiondi**
When running more than 9 transactions at the same time, we get an error: "pool.release is not a function". at line 74 in lib/transaction.js.
I found out that this.pg is returning an EvenEmitter, containing a pool which contains the function release.
Finally, when I call `pool.pool.release` instead of `pool.release`, everything works fine and the connections are released on my pg database.
This modification fixes the issue #109.

### CODE REVIEW

**Comment by slnode** [comment]
Can one of the admins verify this patch? To accept patch and trigger a build add comment ".*ok\W+to\W+test.*"

**Comment by slnode** [comment]
Can one of the admins verify this patch?

**Comment by Elindorath** [comment]
When can this PR could be merged? I think it is a very small and highly valuable one! 😉

**Comment by robinbiondi** [comment]
Hello @raymondfeng. Do you have an idea of when this PR could be processed?
Cheers,

**Comment by slnode** [comment]
Can one of the admins verify this patch?

**Comment by superkhau** [comment]
Can you add some tests to verify your changes and prevent regressions in the future?

In addition, can you rebase with master to get the latest changes on this PR?

**Comment by cadesalaberry** [comment]
Hi there,

**What kind of test do you expect for this PR to be merged ?**


TLDR: `this.pg.release()` does not exist. `this.pg.pool.release()` does.


From what I saw, if you follow the callstack, that's where we [start](https://github.com/robinbiondi/loopback-connector-postgresql/blob/master/lib/postgresql.js#L79):
```js
function PostgreSQL(postgresql, settings) {
   ...
  this.pg = new postgresql.Pool(this.clientConfig);
  this.settings = settings;
  ...
}
```

Down the trail, you see that `Pool` comes [from a poolFactory](https://github.com/brianc/node-postgres/blob/master/lib/index.js#L22) which returns a `pg-pool` module:
```js
PG = function(clientConstructor) {
  EventEmitter.call(this);
  ...
  this.Client = clientConstructor;
  this.Pool = poolFactory(this.Client);
  ...
};
```

That's where the naming is confusing, as the `pg-pool` module also contains a `pool` [object]():
```js
Pool = module.exports = function (options, Client) {
  ...
  this.options = objectAssign({}, options)
  ...
  this.pool = new genericPool.Pool(this.options)
  ...
}
```
[here](https://github.com/brianc/node-pg-pool/blob/master/index.js#L19) if you want to release the pool, you will have to go through the pool object in which there in an actual pool to [release](https://github.com/coopernurse/node-pool/blob/master/lib/generic-pool.js#L406).

**Comment by slnode** [comment]
Can one of the admins verify this patch?

**Comment by slnode** [comment]
Can one of the admins verify this patch?

**Comment by ssh24** [comment]
@robinbiondi Could you please sign the `CLA` so we can proceed with this PR?https://cla.strongloop.com/agreements/strongloop/loopback-connector-postgresql

@slnode test please

**Comment by robinbiondi** [comment]
@ssh24 I just did it.

Thanks!
